Tests/SwiftDocCTests/Model/PossibleValuesTests.swift

```diff
@@ -0,0 +1,168 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+import SymbolKit
+import Foundation
+@testable import SwiftDocC
+import SwiftDocCTestUtilities
+
+
+class PossibleValuesTests: XCTestCase {
+    
+    func testPossibleValuesDiagnostics() throws {
+        // Check that a problem is emitted when extra possible values are documented.
+        var (url, _, context) = try testBundleAndContext(copying: "DictionaryData") { url in
+            try """
+            #  ``Month``
+            
+            Month object.
+            
+            - PossibleValues:
```

Comment:
If I replace this with
```suggestion
            - PossibleValues:
              - January: First links to ``Month``
              - February: Second links to <doc:NotFoundArticle>
              - March: Third links to ``NotFoundSymbol``
              - April: Fourth isn't a known possible value
```
I would expect there to be two more diagnostics about the unresolved links for the second and third values but those are never emitted. Similarly, the link in the first value above, doesn't get resolved.

This is the test that I asked for in[ this comment](https://github.com/swiftlang/swift-docc/pull/857#pullrequestreview-1938903776) in the previous PR.